python3/totoro_agent/utils/util_functions.py

```diff
@@ -363,6 +362,59 @@ def is_movement(action):
     ]
     return action in actions
 
+# Update value map based on reward entities input
+def update_value_map(rval, value_map, world_dim, max_reward_spread=0, OUTER_MAP_VALUES=((-100, -100),(-100, -100))):
+    '''
+    Updates the reward value map with mask matrix application, based on reward entity.
+    Returns a numpy array representing the updated reward value map.
+    '''
+    # Add map padding
+    pad_dim = (world_dim[0]-1,world_dim[1]-1)
+    value_map = np.pad(value_map, (pad_dim,pad_dim), 'constant', constant_values=OUTER_MAP_VALUES)
+    rval_offset = pad_dim[0]-1
+
+    reward = rval[2]
+    reward_discount = reward/abs(reward)
+    rval_offset = 8 # padding offset
+    reward_spread=0
+    
+    # Max reward spread shall not exceed the dimension of the map
+    # (mask matrices of +1 or -1 are applied in additive layers that correspond to the magnitude of the reward value)
+    max_reward_spread = min(max_reward_spread, world_dim[0]-1)
+    
+    if reward > 0:
+        # positive reward value
+        for i, value in enumerate(range(0, reward, 1)):
+            if i <= max_reward_spread:
+                reward_spread = i
+            xstart= rval[1] + rval_offset - reward_spread
+            xend = rval[1] + rval_offset + 1 + reward_spread
+            ystart = rval[0] + rval_offset - reward_spread
+            yend = rval[0] + rval_offset + 1 + reward_spread
+
+            # Updates reward values in the map matrix.
+            value_map[xstart:xend,ystart:yend] = value_map[xstart:xend,ystart:yend] + reward_discount
+            
+    elif reward < 0:
+        # negative reward value
+        for i, value in enumerate(range(0, reward, -1)):
+            if i <= max_reward_spread:
+                reward_spread = i
+            xstart= rval[1] + rval_offset - reward_spread
+            xend = rval[1] + rval_offset + 1 + reward_spread
+            ystart = rval[0] + rval_offset - reward_spread
+            yend = rval[0] + rval_offset + 1 + reward_spread
+
+            # Updates reward values in the map matrix.
+            value_map[xstart:xend,ystart:yend] = value_map[xstart:xend,ystart:yend] + reward_discount
+    else:
+        # Reward assigned is 0.
+        pass
+
+    # Remove map padding
+    value_map = value_map[world_dim[0]-1:world_dim[0]+pad_dim[0],world_dim[0]-1:world_dim[0]+pad_dim[0]]
+
+    return value_map
 
 def get_value_map(world, walls, game_objects, reward_map, pinch_points=None, use_default=True):
```

Comment:
as discussed, need to update offset and orientation

Reply:
Updated as discussed.